lawnchair/src/app/lawnchair/search/data/SearchRequest.kt

```diff
@@ -0,0 +1,193 @@
+package app.lawnchair.search.data
+
+import android.content.Context
+import android.provider.ContactsContract
+import android.provider.MediaStore
+import android.util.Log
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.json.JSONArray
+import org.json.JSONObject
+
+suspend fun getStartPageSuggestions(query: String, max: Int): List<String> {
+    if (query.isEmpty() || query.isBlank() || max <= 0) return emptyList()
+    return withContext(Dispatchers.IO) {
+        try {
+            val url = URL("https://www.startpage.com/suggestions?q=$query&segment=startpage.lawnchair&partner=lawnchair&format=opensearch")
+            val connection = (url.openConnection() as HttpURLConnection).apply { requestMethod = "GET" }
```

Comment:
Use Retrofit instead.

Reply:
Oh yeah, just check we have retrofit will implement that